### PR TITLE
Try wrap read/write ops 

### PR DESCRIPTION
**Pull request by DreamWuGit**
currently in buss mapping op codes,   it is often used` state.push_op(xxx)`, which is not very much meaningful & readable,  
Here try to wrap read/write ops  with specific op naming, like `call_context_read` `memory_read` `stack_write` etc.    this PR just update `sload` `mload` as an example !
Glad to know if you like it !    

if so, we update all op types in further PR, Thanks !

### CODE REVIEW

**Comment by ed255** [comment]
Thanks for this proposal @DreamWuGit !  I think this is a really good idea; and makes the API of the circuit input builder for opcodes a bit more similar to how the constraint builder does the lookups.

Take into account that recently this PR https://github.com/appliedzkp/zkevm-circuits/pull/482 was merged, which splits the circuit input builder into different files; so you'll need to update you PR :)

File: bus-mapping/src/circuit_input_builder.rs [hunk 1]
```diff
@@ -902,6 +923,38 @@ impl<'a> CircuitInputStateRef<'a> {
         Ok(())
     }
 
+    /// Push a [`MemoryOp`] into the [`OperationContainer`] with the next
```
Comment (ChihChengLiang) [inline bus-mapping/src/circuit_input_builder.rs:926]:
```suggestion
    /// Push a read type [`MemoryOp`] into the [`OperationContainer`] with the next
```

File: bus-mapping/src/circuit_input_builder.rs [hunk 2]
```diff
@@ -902,6 +923,38 @@ impl<'a> CircuitInputStateRef<'a> {
...
+        step: &mut ExecStep,
+        address: MemoryAddress,
+        value: u8,
+    ) -> Result<(), Error> {
+        let call_id = self.call()?.call_id;
+        self.push_op(step, RW::READ, MemoryOp::new(call_id, address, value));
+        Ok(())
+    }
+
+    /// Push a [`MemoryOp`] into the [`OperationContainer`] with the next
```
Comment (ChihChengLiang) [inline bus-mapping/src/circuit_input_builder.rs:942]:
```suggestion
    /// Push a write type [`MemoryOp`] into the [`OperationContainer`] with the next
```

File: bus-mapping/src/circuit_input_builder.rs [hunk 3]
```diff
@@ -919,6 +972,38 @@ impl<'a> CircuitInputStateRef<'a> {
         Ok(())
     }
 
+    /// Push a [`StackOp`] into the [`OperationContainer`] with the next
```
Comment (ChihChengLiang) [inline bus-mapping/src/circuit_input_builder.rs:975]:
```suggestion
    /// Push a write type [`StackOp`] into the [`OperationContainer`] with the next
```

File: bus-mapping/src/circuit_input_builder.rs [hunk 4]
```diff
@@ -919,6 +972,38 @@ impl<'a> CircuitInputStateRef<'a> {
...
+        step: &mut ExecStep,
+        address: StackAddress,
+        value: Word,
+    ) -> Result<(), Error> {
+        let call_id = self.call()?.call_id;
+        self.push_op(step, RW::WRITE, StackOp::new(call_id, address, value));
+        Ok(())
+    }
+
+    /// Push a [`StackOp`] into the [`OperationContainer`] with the next
```
Comment (ChihChengLiang) [inline bus-mapping/src/circuit_input_builder.rs:991]:
```suggestion
    /// Push a read type [`StackOp`] into the [`OperationContainer`] with the next
```

**Review by ChihChengLiang — APPROVED** [review]
Added some comments on the documentation.
Otherwise LGTM.

File: bus-mapping/src/circuit_input_builder.rs (hunk 1 above)
Comment (DreamWuGit) [reply bus-mapping/src/circuit_input_builder.rs]: updated in https://github.com/appliedzkp/zkevm-circuits/pull/497/commits/206d5fd8df436421e37e2a80bfa8b2965c487361 thanks

File: bus-mapping/src/circuit_input_builder.rs (hunk 2 above)
Comment (DreamWuGit) [reply bus-mapping/src/circuit_input_builder.rs]: updated in https://github.com/appliedzkp/zkevm-circuits/pull/497/commits/206d5fd8df436421e37e2a80bfa8b2965c487361 thanks

File: bus-mapping/src/circuit_input_builder.rs (hunk 3 above)
Comment (DreamWuGit) [reply bus-mapping/src/circuit_input_builder.rs]: updated in https://github.com/appliedzkp/zkevm-circuits/pull/497/commits/206d5fd8df436421e37e2a80bfa8b2965c487361 thanks

File: bus-mapping/src/circuit_input_builder.rs (hunk 4 above)
Comment (DreamWuGit) [reply bus-mapping/src/circuit_input_builder.rs]: updated in https://github.com/appliedzkp/zkevm-circuits/pull/497/commits/206d5fd8df436421e37e2a80bfa8b2965c487361 thanks

**Comment by DreamWuGit** [comment]
> Thanks for this proposal @DreamWuGit ! I think this is a really good idea; and makes the API of the circuit input builder for opcodes a bit more similar to how the constraint builder does the lookups.
> 
> Take into account that recently this PR #482 was merged, which splits the circuit input builder into different files; so you'll need to update you PR :)

@ed255  rebased !

**Comment by ed255** [comment]
@DreamWuGit would you like to extend this PR to introduce the missing wrappers and to use them in all opcode `gen_associated_ops`?  Or would you prefer to get this PR merged and extend this concept in a future PR?

**Comment by DreamWuGit** [comment]
> @DreamWuGit would you like to extend this PR to introduce the missing wrappers and to use them in all opcode `gen_associated_ops`? Or would you prefer to get this PR merged and extend this concept in a future PR?

@ed255  I prefer do it in future PR as mentioned in the first comments ! :)

**Review by ed255 — APPROVED** [review]
LGTM!

**Pull request merged**